helpers/sql-obfuscation/test/helpers/sql_obfuscation_test.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Helpers::SqlObfuscation do
+  let(:obfuscation_limit) { 2000 }
+  let(:adapter) { :default }
+  let(:sql) { "SELECT * from users where users.id = 1 and users.email = 'test@test.com'" }
+  let(:expected) { 'SELECT * from users where users.id = ? and users.email = ?' }
+  let(:obfuscate_sql) do
+    OpenTelemetry::Helpers::SqlObfuscation.obfuscate_sql(
+      sql,
+      obfuscation_limit: obfuscation_limit,
+      adapter: adapter
+    )
+  end
+
+  describe '.obfuscate_sql' do
+    it 'returns an obfuscated sql statement' do
+      assert_equal(expected, obfuscate_sql)
+    end
+
+    describe 'when named args with defaults are not passed' do
+      let(:obfuscate_sql) { OpenTelemetry::Helpers::SqlObfuscation.obfuscate_sql(sql) }
+
+      it 'obfuscates the SQL' do
+        assert_equal(expected, obfuscate_sql)
+      end
+    end
+
+    describe 'when sql has marginalia-style prepended comments' do
+      let(:sql) do
+        '/*application:web,controller:blob,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,' \
+          'endpoint_id:Projects::BlobController#show*/ SELECT "routes".* FROM "routes" WHERE "routes"' \
+          '."source_id" = 75 AND "routes"."source_type" = \'Namespace\' LIMIT 1'
+      end
+
+      let(:expected) { '? SELECT ?.* FROM ? WHERE ?.? = ? AND ?.? = ? LIMIT ?' }
```

Comment:
This was ... unexpected. Makes sense, but annoying if any generated SQL queries quote table or column names. Does AR do this or is this just an invented (but valid) example?